controllers/datamover.go

```diff
@@ -164,6 +166,61 @@ func (r *DPAReconciler) ReconcileDataMoverController(log logr.Logger) (bool, err
 	return true, nil
 }
 
+func (r *DPAReconciler) ReconcileDataMoverConfigs(log logr.Logger) (bool, error) {
+
+	// fetch latest DPA instance
+	dpa := oadpv1alpha1.DataProtectionApplication{}
+	if err := r.Get(r.Context, r.NamespacedName, &dpa); err != nil {
+		return false, err
+	}
+
+	// check configMap already exists
+	confMap, confMapExists, err := r.checkConfigMapExists()
+	if err != nil {
+		return false, err
+	}
+
+	// check for existing configMap but no data mover configs set
+	if !r.checkDataMoverConfigs(&dpa) && confMapExists {
+
+		err := r.Delete(context.Background(), confMap, &client.DeleteOptions{})
+		if err != nil {
+			return false, err
+		}
+	}
+
+	// create configmap only if data mover is enabled and has config values
+	if r.checkIfDataMoverIsEnabled(&dpa) && r.checkDataMoverConfigs(&dpa) {
+
+		// create configmap to pass values to data mover CRs
+		cm := corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      DataMoverConfigMapName,
+				Namespace: dpa.Namespace,
+			},
+		}
+
+		op, err := controllerutil.CreateOrPatch(r.Context, r.Client, &cm, func() error {
+			cm.Data = r.makeConfigMapData(&dpa)
```

Comment:
Maybe follow the `build<ResourceName>` pattern here as well as we have done through the OADP Operator tasks where we pass on the resource pointer and build its specs